File: SMLanguage/Models/Game.cs
using System.Collections.Generic;

namespace SMLanguage.Models
{
    public class Game
    {
        public ImageFormatType ImageFormatType { get; set; }

        public string Title { get; set; }

        public bool DebugMode { get; set; }

        public string StartFile { get; set; }

        public string ScenarioExtension { get; set; }

        public RunningScenario CurrentScenario { get; set; }

        public List<RanScenario> PreviousScenarios { get; set; }

    }
}

File: SM-engine/src/models/Game.ts
import { ImageFormatType } from '../enums';
import { RunningScenario } from './RunningScenario';
import { RanScenario } from './RanScenario';

export class Game {
  public ImageFormatType: ImageFormatType = ImageFormatType.PNG;
  public Title: string = '';
  public TitleScreenImageName: string = '';
  public DebugMode: boolean = false;
  public StartFile: string = '';
  public ScenarioExtension: string = '';
  public CurrentScenario: RunningScenario | null = null;
  public PreviousScenarios: RanScenario[] = [];
}
